pkg/driver/node.go

```diff
@@ -512,17 +545,62 @@ func (ns *node) NodeUnpublishVolume(
 		return nil, status.Error(codes.InvalidArgument, "Target path not provided")
 	}
 
+	if ok := ns.volumeTransition.Insert(volumeID); !ok {
+		msg := fmt.Sprintf("an operation on this volume=%q is already in progress", volumeID)
+		return nil, status.Error(codes.Aborted, msg)
+	}
+
+	defer func() {
+		logrus.Infof("NodeUnPublishVolume: volume: {%q} operation finished", volumeID)
+		ns.volumeTransition.Delete(volumeID)
+	}()
+
+	if err := ns.unmount(volumeID, target); err != nil {
+		return nil, err
+	}
+
+	return &csi.NodeUnpublishVolumeResponse{}, nil
+}
+
+func (ns *node) isAlreadyMounted(volID, path string) error {
+	currentMounts := map[string]bool{}
+	mountList, err := ns.mounter.List()
+	if err != nil {
+		return fmt.Errorf("Failed to list mount paths")
+	}
+
+	for _, mntInfo := range mountList {
+		if strings.Contains(mntInfo.Path, volID) {
+			currentMounts[mntInfo.Path] = true
+		}
+	}
+
+	// if volume is mounted at more than one place check if this request is
+	// for the same path that is already mounted. Return nil if the path is
+	// mounted already else return err so that it gets unmounted in the
+	// next subsequent calls in respective rpc calls (NodeUnpublishVolume, NodeUnstageVolume)
+	if len(currentMounts) > 1 {
```

Comment:
even if it is mounted at one place, shouldn't we verify in that case also that it is mounted at the corresponding targetpath or not? it should len(currentMounts) > 0?

Reply:
can you test a Deployment with two replica and see if it is working.

Reply:
yes, you are right but we are not supporting RWM.

Reply:
please make a note of it that we don't allow to mount it more than once on a node.